packages/cli/src/lib/implementation/only-plugins.utils.ts

```diff
@@ -1,65 +1,27 @@
 import chalk from 'chalk';
-import type { CategoryConfig, CoreConfig } from '@code-pushup/models';
-import { ui } from '@code-pushup/utils';
-
-export function filterPluginsBySlug(
-  plugins: CoreConfig['plugins'],
-  { onlyPlugins }: { onlyPlugins?: string[] },
-): CoreConfig['plugins'] {
-  if (!onlyPlugins?.length) {
-    return plugins;
-  }
-  return plugins.filter(plugin => onlyPlugins.includes(plugin.slug));
-}
-
-// skip the whole category if it has at least one skipped plugin ref
-// see https://github.com/code-pushup/cli/pull/246#discussion_r1392274281
-export function filterCategoryByPluginSlug(
-  categories: CategoryConfig[],
-  {
-    onlyPlugins,
-    verbose = false,
-  }: { onlyPlugins?: string[]; verbose?: boolean },
-): CategoryConfig[] {
-  if (categories.length === 0) {
-    return categories;
-  }
-  if (!onlyPlugins?.length) {
-    return categories;
-  }
-
-  return categories.filter(category =>
-    category.refs.every(ref => {
-      const isNotSkipped = onlyPlugins.includes(ref.plugin);
-
-      if (!isNotSkipped && verbose) {
-        ui().logger.info(
-          `${chalk.yellow('⚠')} Category "${
-            category.title
-          }" is ignored because it references audits from skipped plugin "${
-            ref.plugin
-          }"`,
-        );
-      }
-
-      return isNotSkipped;
-    }),
-  );
-}
+import type { CategoryConfig, PluginConfig } from '@code-pushup/models';
+import { filterItemRefsBy, ui } from '@code-pushup/utils';
 
 export function validateOnlyPluginsOption(
-  plugins: CoreConfig['plugins'],
   {
-    onlyPlugins,
+    plugins,
+    categories,
+  }: {
+    plugins: PluginConfig[];
+    categories: CategoryConfig[];
+  },
+  {
+    onlyPlugins = [],
     verbose = false,
-  }: { onlyPlugins?: string[]; verbose?: boolean },
+  }: { onlyPlugins?: string[]; verbose?: boolean } = {},
 ): void {
-  const missingPlugins = onlyPlugins?.length
-    ? onlyPlugins.filter(plugin => !plugins.some(({ slug }) => slug === plugin))
-    : [];
+  const onlyPluginsSet = new Set(onlyPlugins);
+  const missingPlugins = onlyPlugins.filter(
+    plugin => !plugins.some(({ slug }) => slug === plugin),
+  );
 
   if (missingPlugins.length > 0 && verbose) {
-    ui().logger.warning(
+    ui().logger.info(
```

Comment:
Shouldn't this be a warning? It's a mistake by the user.

Reply:
This was on purpose. loggin warnings logs into `stderr` instead of `stdout`. Changed it on multiple places.

Reply:
Warnings logged to stderr is a good thing, though.

Reply:
But that's out of scope of this PR.